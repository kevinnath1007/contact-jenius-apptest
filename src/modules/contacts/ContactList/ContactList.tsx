import React from 'react';
import {View, FlatList, Text} from 'react-native';

import ContactListItem from './fragments/ContactListItem';
import {ContactListType} from './contactList.api';
import {getContactList} from './contactList.action';
import {useSelector} from 'react-redux';
import useAppDispatch from '../../../hooks/useAppDispatch';

const ContactList: React.FC = () => {
  const contactListData = useSelector(state => state.contactList.value);
  const dispatch = useAppDispatch();

  React.useEffect(() => {
    dispatch(getContactList());
  }, []);
  const renderItem = ({firstName, lastName, photo}: ContactListType) => {
    return (
      <ContactListItem
        firstName={firstName}
        lastName={lastName}
        photo={photo}
      />
    );
  };

  return (
    <View>
      <Text>test</Text>
      <FlatList data={[]} renderItem={renderItem} />
    </View>
  );
};

export default ContactList;
