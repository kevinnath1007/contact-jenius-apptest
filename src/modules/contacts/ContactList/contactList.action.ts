import {actionIds, BaseAction} from '../../../common';
import {ContactListType} from './contactList.api';

export const getContactListAction = (contactList: ContactListType): BaseAction => ({
  type: actionIds.GET_CONTACT_LIST,
  payload: contactList,
});
