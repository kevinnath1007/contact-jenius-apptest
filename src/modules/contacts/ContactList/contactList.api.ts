import {coreAPI} from '../../../libraries/network';
import {CoreClientResponse} from '../../../libraries/network/instance/core/responseInterceptor';

export type ContactListType = {
  id: string;
  firstName: string;
  lastName: string;
  age: number;
  photo: string;
};

export const getContactList = async () => {
  const response = await coreAPI.get<
    null,
    CoreClientResponse<ContactListType[]>
  >('/contact');

  if (response.ok) {
    return response.data;
  } else if (__DEV__) {
    console.log(response.message);
  }
};
