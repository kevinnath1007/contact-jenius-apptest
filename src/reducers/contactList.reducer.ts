import {BaseAction, actionIds} from '../common';

export type ContactListState = number[];

export const contactListReducer = (
  state: ContactListState = [0],
  action: BaseAction,
) => {
  switch (action.type) {
    case actionIds.GET_CONTACT_LIST:
      return [...state, action.payload];
  }
  return state;
};
