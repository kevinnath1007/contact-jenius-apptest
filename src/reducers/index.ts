import {
  combineReducers,
  configureStore,
  applyMiddleware,
} from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';

import {contactListReducer, ContactListState} from './contactList.reducer';

export interface State {
  contactList: ContactListState;
}

const rootReducers = combineReducers<State>({
  contactList: contactListReducer,
});

export type RootState = ReturnType<typeof rootReducers>;

const sagaMiddleware = createSagaMiddleware();
export const store = configureStore({
  reducer: rootReducers,
});

export type AppDispatch = typeof store.dispatch;
