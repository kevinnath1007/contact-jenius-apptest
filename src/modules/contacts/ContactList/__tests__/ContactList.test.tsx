import React from 'react';
import {Provider} from 'react-redux';
import ContactList from '../ContactList';
import {
  // fireEvent,
  render,
} from '@testing-library/react-native';
import {store} from '../../../../reducers';

describe('ContactList', () => {
  const navigationMock = {
    navigate: jest.fn(),
    goBack: jest.fn(),
    setOptions: jest.fn(),
  };

  const Component = () =>
    render(
      <Provider store={store}>
        {/*@ts-ignore*/}
        <ContactList navigation={navigationMock} route={{}} />
      </Provider>,
    );

  test('snapshot: component match', () => {
    const {toJSON} = Component();

    expect(toJSON()).toMatchSnapshot();
  });

  // test('function: component match', () => {
  //   const {getByTestId} = Component();
  //
  //   fireEvent.press(getByTestId(''));
  //   expect(navigationMock.navigate).toBeCalled();
  //   expect(navigationMock.navigate).toBeCalledTimes(1);
  // });
});
