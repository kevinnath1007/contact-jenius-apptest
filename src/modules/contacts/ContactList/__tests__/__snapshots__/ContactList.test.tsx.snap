// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList snapshot: component match 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
